packages/tools/buildServerDocker.ts

```diff
@@ -44,10 +55,44 @@ async function main() {
 	console.info('isPreRelease:', isPreRelease);
 	console.info('Docker tags:', dockerTags.join(', '));
 
-	await execCommand2(`docker build -t "joplin/server:${imageVersion}" ${buildArgs} -f Dockerfile.server .`);
-	for (const tag of dockerTags) {
-		await execCommand2(`docker tag "joplin/server:${imageVersion}" "joplin/server:${tag}"`);
-		if (pushImages) await execCommand2(`docker push joplin/server:${tag}`);
+	const platforms = [
+		'linux/amd64',
+		'linux/arm64',
+		'linux/arm/v7',
+	];
+
+	// this will build a bunch of local image tags named: ${imageVersion}-${platform} with the slashes replaced with dashes
+	for (const platform of platforms) {
+		const normalizedPlatform = normalizePlatform(platform);
+		await execCommand2([
+			'docker', 'build',
+			'--platform', platform,
+			'-t', `${DockerImageName}:${imageVersion}-${normalizedPlatform}`,
+			...flattenedBuildArgs,
+			'-f', 'Dockerfile.server',
+			'.',
+		]);
+		if (pushImages) {
+			await execCommand2([
+				'docker', 'push', `${DockerImageName}:${imageVersion}-${normalizedPlatform}`,
+			]);
+		}
+	}
+
+	// now we have to create the right manifests and push them
+	if (pushImages) {
+		for (const tag of dockerTags) {
+			// manifest create requires the tags being amended in to exist on the remote, so this all can only happen if pushImages is true
+			await execCommand2([
+				'docker', 'manifest', 'create',
+				`${DockerImageName}:${tag}`,
+				...flatten(platforms.map((platform) => ['--amend', `${DockerImageName}:${imageVersion}-${normalizePlatform(platform)}`])),
```

Comment:
I feel this flatten function makes the code harder to read, at least for me. Can't it be done without it?